euphonic/spectra.py

```diff
@@ -1656,6 +1780,149 @@ def from_dict(cls: Type[T], d: Dict[str, Any]) -> T:
+
+    Intended for convenient storage of contributions to spectral maps such as
+    S(Q,w). This object can be indexed or iterated to obtain individual
+    Spectrum2D.
+
+    Attributes
+    ----------
+    x_data
+        Shape (n_x_data,) or (n_x_data + 1,) float Quantity. The x_data
+        points (if size == (n_x_data,)) or x_data bin edges (if size
+        == (n_x_data + 1,))
+    y_data
+        Shape (n_y_data,) or (n_y_data + 1,) float Quantity. The y_data
+        points (if size == (n_y_data,)) or y_data bin edges (if size
+        == (n_y_data + 1,))
+    z_data
+        Shape (n_entries, n_x_data, n_y_data) float Quantity. The spectral data
+        in x and y, indexed over components
+    x_tick_labels
+        Sequence[Tuple[int, str]] or None. Special tick labels e.g. for
+        high-symmetry points. The int refers to the index in x_data the
+        label should be applied to
+    metadata
+        Dict[str, Union[int, str, LineData]] or None. Contains metadata
+        about the spectra. Keys should be strings and values should be
+        strings or integers.
+        There are some functional keys:
+
+          - 'line_data' : LineData
+                          This is a Sequence[Dict[str, Union[int, str]],
+                          it contains metadata for each spectrum in
+                          the collection, and must be of length
+                          n_entries
+    """
+
+    # Private attributes used by SpectrumCollectionMixin
+    _bin_axes = ("x", "y")
+    _spectrum_axis = "z"
+    _item_type = Spectrum2D
+
+    def __init__(
+            self, x_data: Quantity, y_data: Quantity, z_data: Quantity,
+            x_tick_labels: Optional[Sequence[Tuple[int, str]]] = None,
+            metadata: Optional[Metadata] = None
+    ) -> None:
+        _check_constructor_inputs(
+            [z_data, x_tick_labels, metadata],
+            [Quantity, [list, type(None)], [dict, type(None)]],
+            [(-1, -1, -1), (), ()],
+            ['z_data', 'x_tick_labels', 'metadata'])
+        nx = z_data.shape[1]
+        ny = z_data.shape[2]
+        _check_constructor_inputs(
+            [x_data, y_data],
+            [Quantity, Quantity],
+            [[(nx,), (nx + 1,)], [(ny,), (ny + 1,)]],
+            ['x_data', 'y_data'])
+
+        self._set_data(x_data, 'x')
+        self._set_data(y_data, 'y')
+        self.x_tick_labels = x_tick_labels
+        self._set_data(z_data, 'z')
+        if metadata and 'line_data' in metadata.keys():
+            if len(metadata['line_data']) != len(z_data):
+                raise ValueError(
+                    f'z_data contains {len(z_data)} spectra, but '
+                    f'metadata["line_data"] contains '
+                    f'{len(metadata["line_data"])} entries')
+        self.metadata = {} if metadata is None else metadata
+
+    def _split_by_indices(self, indices: Sequence[int] | np.ndarray
```

Comment:
Classmethod?

Reply:
I don't see how that could work. 1D version needs access to x_data and y_data, 2D version needs access to x_data, y_data and z_data. Both are called by a common `split()` method from parent class.

Reply:
I just saw the `type(self)` which immediately makes me go "Classmethod?" possibly a bad reflex, but just to think about.